holonote/annotate/table.py

```diff
@@ -337,5 +319,29 @@ def load_annotation_table(self, conn: Connector, fields: list[str], spec: SpecDi
 
         self._region_df = pd.concat(data, ignore_index=True)
 
-        self._update_index()
         self.clear_edits()
+
+    @property
+    def region_df(self):
```

Comment:
This seems to be the core of this PR - makes sense to me!

The overall PR diff would be simpler if you kept these two properties private (i.e. `_region_df` and `_field_df`): I'm not sure these are concepts users need to think about.

Reply:
For clarity, I would consider keeping the old names to make this PR clean and I would be happy to discuss whether these should be private/public names in a separate PR.

Just a thought: no need to follow this suggestion if you don't agree (not trying to give you unnecessary work!)

Reply:
Correct. I am postponing the concatenating of `_region_df` and `_field_df`. These changes have no effect when adding a single item, but a significant change when adding many in succession as can be seen by the benchmark.